src/base/lane.cc

```diff
@@ -44,8 +44,24 @@ double Lane::do_length() const { return lane_geometry_->ArcLength(); }
 maliput::api::RBounds Lane::do_lane_bounds(double s) const { return lane_geometry_->RBounds(s); }
 
 maliput::api::RBounds Lane::do_segment_bounds(double s) const {
-  // TODO: Implement
-  MALIPUT_THROW_MESSAGE("Not implemented");
+  const double bound_left = ComputeDistanceToSegmentBoundary(kToLeft, s);
+  const double bound_right = ComputeDistanceToSegmentBoundary(kToRight, s);
+  const double tolerance = lane_geometry_->linear_tolerance();
+  return {bound_right < tolerance ? -tolerance : -bound_right, bound_left < tolerance ? tolerance : bound_left};
+}
+
+double Lane::ComputeDistanceToSegmentBoundary(bool to_left, double s) const {
```

Comment:
Is this function something that we can push to maliput::geometry_base::Lane or that namespace in particular? I think that the only specific class state is the reference to `this` which can easily be solved adding one argument. The other implementation specific data is the conversion between s-coordinates of multiple lanes. I think we can encapsulate that in a functor or by template strategy injection.

What do you think? We can potentially remove duplicated code with a similar approach.

Reply:
Agree. I think that the algorithm used here and[ in maliput_malidrive ](https://github.com/maliput/maliput_malidrive/blob/814882a828feded7949e51d7eeef6414a2b701f5/src/maliput_malidrive/base/lane.cc#L133)is the same. The only thing changing is the "strategy" to get the scale from one lane to the other.
I think that it'd fit better as a helper method at `geometry_base` rather than being part of the geometry_base::Lane's interface.

This would affect maliput and maliput_malidrive. For the purpose of not blocking this PR. Let me create an issue and tackle this later on.

Reply:
https://github.com/maliput/maliput/issues/523